app/pages/lab/iiif/components/hooks/useSubjectUploads.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from 'react'
+
+import { createSubject } from '../helpers'
+
+export default function useSubjectUploads(snapshots = [], subjectSet) {
+  const [uploaded, setUploaded] = useState([])
+  const [failed, setFailed] = useState([])
+  const [finished, setFinished] = useState(false)
+  const [loaded, setLoaded] = useState(false)
+  const [uploadCount, setUploadCount] = useState(0)
+
+  async function uploadSubject(snapshot) {
+    try {
+      const panoptesSubject = await createSubject(snapshot)
+      setUploaded(subjects => [...subjects, panoptesSubject])
+      setUploadCount(count => count + 1)
+    } catch (error) {
+      console.error(error)
+      setFailed(subjects => [...subjects, snapshot])
```

Comment:
Automatic retry on failed uploads = solid 👍 

But am I right in that there's no limit to how often it'll retry to upload a failed attempt? If so, I'll need to test if there's any way this could result in inadvertent API spam. (Also need to see if the upload process can be manually stopped mid-way and what the effects are.)

Reply:
You’re right. This will retry forever if your network connection dies, for example. Reloading the page will probably kill it. 

I can add a Cancel button, to stop the upload and delete the subject set.

Reply:
A Cancel button is definitely a nice-to-have 👍 I wouldn't let that hold up this PR (if adding the Cancel button is a lot of work) though, since the PR is looking really good as is.